src/OpenLoco/Paint/PaintEntity.cpp

```diff
@@ -0,0 +1,99 @@
+#include "PaintEntity.h"
+#include "../Config.h"
+#include "../Interop/Interop.hpp"
+#include "../Map/Tile.h"
+#include "../Things/Misc.h"
+#include "../Things/ThingManager.h"
+#include "Paint.h"
+
+using namespace OpenLoco::Interop;
+using namespace OpenLoco::Ui::ViewportInteraction;
+
+namespace OpenLoco::Paint
+{
+    // 0x004B0CCE
+    void paintVehicleEntities(PaintSession& session, vehicle_base* base)
+    {
+        registers regs{};
+        regs.ax = base->x;
+        regs.cx = base->y;
+        regs.dl = base->z;
+        regs.ebx = (base->sprite_yaw + (session.rotation() << 4)) & 0x3F;
```

Comment:
Is the anding neccesary? Wil rotation() ever be higher than zero?

Reply:
It needs to roll over correctly when the entity has been rotated. So yes it is required

Reply:
Maybe have getRotation() return (_rotation & 0b11)?

Reply:
But I'd need a set as well. The sprite yaw is a bit of a special rotation field.